rust/kernel/fs.rs

```diff
@@ -78,6 +93,178 @@ pub mod ro {
         }
     }
 
+    /// A file system super block.
+    ///
+    /// Wraps the kernel's `struct super_block`.
+    #[repr(transparent)]
+    pub struct SuperBlock<T: Type + ?Sized>(Opaque<bindings::super_block>, PhantomData<T>);
+
+    /// Required superblock parameters.
+    ///
+    /// This is used in [`NewSuperBlock::init`].
+    pub struct SuperParams {
+        /// The magic number of the superblock.
+        pub magic: u32,
+
+        /// The size of a block in powers of 2 (i.e., for a value of `n`, the size is `2^n`).
+        pub blocksize_bits: u8,
+
+        /// Maximum size of a file.
+        ///
+        /// The maximum allowed value is [`super::MAX_LFS_FILESIZE`].
+        pub maxbytes: i64,
```

Comment:
what happens when the maximum is exceeded?

Reply:
I meant the latter, is that architecture dependent? If yes, we should add a (static?) check for that. maybe also add it regardless, since it doesn't hurt (of course only if it does not produce a warning)?